### ISSUE TITLE
"Range Added" has incorrect units on the Charges dashboard

### ISSUE THREAD

**Issue by JakobLichterfeld** [teslamate-org/teslamate#4348]
> Hello, I'm a new user in this exact issue is something that bothered me enough to come here and search for answers. Was this issue corrected yet? I'm see the same behavior as op. 
> 
> In the "range added" column, I would expect to see the number of miles that I gained during the charging session. Instead, it shows me how _fast_ the charging occurred. This is super misleading because if I charged for 28 min and gained 20 miles of range, it displays 45 mph. If I was charged for exactly one hour, then that value would be accurate but I'd argue that most people are not charging for exactly one hour, and would rather see the number of miles worth of change during the amount of time they were plugged in and not for one hour. 
> 
> Can this please be corrected or can there be an option to display the actual range added?

_Originally posted by @1920x1200 in https://github.com/teslamate-org/teslamate/issues/4087#issuecomment-2453285969_

**Comment by JakobLichterfeld** [comment]
@1920x1200 welcome to TeslaMate, nice to have you.

Please do not report an issue in a closed issue, it was only by accident I had notifications on for the closed issue.

Regarding the issue. @DrMichael solved it by renaming it to Range added / h, but @jheredianet accidentally reverted it when adding the Summary in #4144.

@swiffer What's your opinion on this? I personally do not see any benefit in Range added / h at all, but some do and would like to add it to MQTT as well #4130

**Comment by swiffer** [comment]
I think it's the unit Tesla is showing during Super Charging as well on screen, don't they. I would therefore say let's keep it like that.

see here, down on bottom of video, KPI in the middle: https://www.tesla.com/fr_ch/blog/introducing-v3-supercharging

`At this rate, a Model 3 Long Range operating at peak efficiency can recover up to 75 miles of charge in 5 minutes and charge at rates of up to 1,000 miles per hour.`

So therefore, rename the column to "Charge Rate" instead of "Range added" ?

**Comment by JakobLichterfeld** [comment]
> I think it's the unit Tesla is showing during Super Charging as well on screen, don't they. I would therefore say let's keep it like that.

Yeah it is.

> So therefore, rename the column to "Charge Rate (avg)" instead of "Range added" ?

I do like it. A better name than range added / h

**Comment by swiffer** [comment]
Ok, PR open in a sec ;)

**Comment by swiffer** [comment]
I wonder if we should replace Driven (km) and Odometer (km) by Range added instead. Are those 2 really helping here?

**Comment by JakobLichterfeld** [comment]
> I wonder if we should replace Driven (km) and Odometer (km) by Range added instead. Are those 2 really helping here?

Odometer was recently added due to community request: https://github.com/teslamate-org/teslamate/discussions/4109#discussioncomment-10201456

**Comment by swiffer** [comment]
still voting to replace Driven with Range added then, Driven is hard to grasp as it is not really related to the charge

**Comment by DrMichael** [comment]
> So therefore, rename the column to "Charge Rate (avg)" instead of "Range added" ?

Appears clearer. We could also name the column just "km /hr" and "mi / hr". (When looking at the screenshot.

**Comment by DrMichael** [comment]
> still voting to replace Driven with Range added then, Driven is hard to grasp as it is not really related to the charge

Aren't that different things? Driven means distance driven since last charge, right?

I think, Odometer is fine. But yes, Driven and Odometer cover the same issue.

**Comment by JakobLichterfeld** [comment]
Agree, what do you think @DrMichael ?

**Comment by 1920x1200** [comment]
If we're going to keep the _speed_ of charging (in MPH), is there a way to add a column for the number of miles added during a charging session? Personally, I would find that more useful.

**Comment by swiffer** [comment]
That's what I suggested - will open a PR later

**Comment by 1920x1200** [comment]
Thank you for fixing this in the latest update!

**Comment by zexpe** [comment]
>  I myself finding driven not really helpful in context of charges, that's why I wanted to replace it.

I've only just noticed that the Driven column was removed from the Charges tables - I had been on an earlier version until the breaking changes forced me to upgrade to version 4.

I used to find this was a helpful data point - it is a stat that is shown on the dashboard of the car, and is something I like to look at over time - how far did I really manage to drive between charges? 

Does it exist elsewhere in the standard Grafana dashboards for TeslaMate? As in a similar full table over a long time range where I can sort by Driven and compare against start / end SoCs?
